Allow disabling window snapping

akosfekete opened the pull request:
Add disable_window_snap config option as mentioned in #561 

I would update the config wiki page as below.

# Snapping Behaviour
If snapping is enabled, floating windows are snapped (repositioned to a tile) when the window  is dragged close to a Workspace edge. This might lead to undesired behaviour when dragging windows near the edge of the screen, so it is disabled by default.

Snapping can be disabled with `disable_window_snap`:

Default: `disable_window_snap = true`

Example: `disable_window_snap = false` (enables snapping)

hertg commented:
Thanks for submitting :)
Fairly straight-forward PR, I don't have any objections to the changes.
I would even be in favor of defaulting the setting to `true`, because I don't see anyone really using the feature. It is more of an annoyance in its current state. But maybe I'm wrong, are people using it?

@VuiMuich @AethanFoot Thoughts?

VuiMuich commented:
I agree with defaulting to `true` in this case even though it is an actual breaking change, but I would expect the support requests for this change would still be outnumbered by the requests to have this feature at all.

@akosfekete would you mind pushing this little change? When we get no veto against this new default I'll merge tomorrow.

akosfekete commented:
I updated the default value and changed the doc template. I also think defaulting to true makes sense, I was confused by this feature when I started using leftwm (maybe this is more useful when using multiple workspaces on ultrawide monitors?)

juanCortelezzi commented:
The `disable_window_snap` option is enabled by default but I still have some snapping action going on when a floating window is near the border of the screen. I believe this [line](https://github.com/leftwm/leftwm/blob/7b2a1b79ba43484d64383ea6de52ae32cb97bf24/leftwm-core/src/handlers/window_move_handler.rs#L26) might be my issue. Since I use a single monitor, therefore a single workspace I would like to disable this feature and prevent the snapping. Is there any way to do this at the moment? thanks!

VuiMuich commented:
Which build hash are you on? I cant reproduce with my local branch that i forked from main last week (to be precise I am slightly ahead of #793, built with the `ron-config` feature) and I am also on a single monitor.

juanCortelezzi commented:
`LeftWM 0.3.0, Git-Hash: eaf531d` - downloaded from the AUR, also this is [my config](https://github.com/juanCortelezzi/Dotfiles/blob/main/leftwm/.config/leftwm/config.toml).


https://user-images.githubusercontent.com/57237705/180614156-e427ecd8-5752-4c30-b0d3-66c8c8b42a15.mp4

VuiMuich commented:
NP, afaik the AUR "release" only gets bumped, when we update the crate as well. But I believe there is a `leftwm-git` package as well.
